src/platform/PICO/bus_spi_pico.c

```diff
@@ -321,39 +328,52 @@ void spiInternalResetStream(dmaChannelDescriptor_t *descriptor)
 bool spiInternalReadWriteBufPolled(SPI_TypeDef *instance, const uint8_t *txData, uint8_t *rxData, int len)
 {
     // TODO optimise with 16-bit transfers as per stm bus_spi_ll code
-    int bytesProcessed = spi_write_read_blocking(SPI_INST(instance), txData, rxData, len);
+    int bytesProcessed = 0;
+    if (txData && rxData) {
+        bytesProcessed = spi_write_read_blocking(SPI_INST(instance), txData, rxData, len);
+    } else if (txData) {
+        bytesProcessed = spi_write_blocking(SPI_INST(instance), txData, len);
+    } else if (rxData) {
+        // NB tx data: "Generally this can be 0, but some devices require a specific value here, e.g. SD cards expect 0xff" (pico-sdk spi.c).
+        uint8_t repeated_tx_data = 0xff; // cf. dummyTxByte in stm bus_spi_ll.c and for DMA here
+        bytesProcessed = spi_read_blocking(SPI_INST(instance), repeated_tx_data, rxData, len);
+    } else {
+        bprintf("\n*** unexpected spiInternalReadWriteBufPolled with no tx, rx");
+    }
+
     return bytesProcessed == len;
 }
 
-// Initialise DMA before first segment transfer
-void spiInternalInitStream(const extDevice_t *dev, bool preInit)
+void spiInternalInitStream(const extDevice_t *dev, volatile busSegment_t *segment)
 {
 #ifndef USE_DMA
     UNUSED(dev);
-    UNUSED(preInit);
+    UNUSED(segment);
 #else
-    UNUSED(preInit);
-
     busDevice_t *bus = dev->bus;
+    spi_inst_t *spi = SPI_INST(bus->busType_u.spi.instance);
```

Comment:
Not related directly, but:

https://github.com/betaflight/betaflight/blob/90d2503274363000de7b156eecfb2b878302e4a5/src/platform/PICO/include/platform/platform.h#L68-L69

Why is SPI_Typedef declared as SPI0_Type?  With PICO hal, it seems cleaner to  `typedef spi_inst_t SPI_TypeDef;` and use spi0_hw instead of SPI0

Reply:
This is starting to cause trouble - see GD32f4 PR.

One possible way to handle this is to use `SPI(1)` instead of `SPI1`. `SPI(n)` macro can handle all necessary platform-dependent subtletties, returning correctly typed value. 

It should work fine for PICO, but it will be only partial fix for GD